cli/command/image/build.go

```diff
@@ -344,9 +360,18 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 	// Everything worked so if -q was provided the output from the daemon
 	// should be just the image ID and we'll print that to stdout.
 	if options.quiet {
-		fmt.Fprintf(dockerCli.Out(), "%s", buildBuff)
+		imageID = fmt.Sprintf("%s", buildBuff)
+		fmt.Fprintf(dockerCli.Out(), "%s", imageID)
```

Comment:
Oops, I got confused about the context of this code wrong, you are right.

I need to use the first variant you give so that when talking to an older server (which doesn't provide the aux messages) the code in the `options.quiet` will update `imageID` for use in the following `options.imageIDFile` block (this makes `-q --iidfile=FOO` work even when talking to an older daemon).